js/background.js

```diff
@@ -483,40 +499,45 @@
         });
     }
 
-    function onMessageReceived(ev) {
-        var data = ev.data;
-        if (data.message.flags & textsecure.protobuf.DataMessage.Flags.PROFILE_KEY_UPDATE) {
-            var profileKey = data.message.profileKey.toArrayBuffer();
-            return ConversationController.getOrCreateAndWait(data.source, 'private').then(function(sender) {
-              return sender.setProfileKey(profileKey).then(ev.confirm);
-            });
-        }
-        var message = initIncomingMessage(data);
-
-        return isMessageDuplicate(message).then(function(isDuplicate) {
-            if (isDuplicate) {
-                console.log('Received duplicate message', message.idForLogging());
-                ev.confirm();
-                return;
-            }
+  /* eslint-enable */
+  /* jshint ignore:start */
+  async function onMessageReceived(ev) {
+    const { data } = ev;
+
+    // eslint-disable-next-line no-bitwise
+    if (data.message.flags & textsecure.protobuf.DataMessage.Flags.PROFILE_KEY_UPDATE) {
+      const profileKey = data.message.profileKey.toArrayBuffer();
+      const sender = ConversationController.getOrCreateAndWait(data.source, 'private');
+      await sender.setProfileKey(profileKey);
+      // TODO: Is `ev.confirm` a `Promise`? Original code returned it:
+      return ev.confirm();
+    }
 
-            var type, id;
-            if (data.message.group) {
-                type = 'group';
-                id = data.message.group.id;
-            } else {
-                type = 'private';
-                id = data.source;
-            }
+    const message = initIncomingMessage(data);
+    const isDuplicate = await isMessageDuplicate(message);
 
-            return ConversationController.getOrCreateAndWait(id, type).then(function() {
-                return message.handleDataMessage(data.message, ev.confirm, {
-                    initialLoadComplete: initialLoadComplete
-                });
-            });
-        });
+    if (isDuplicate) {
+      console.log('Received duplicate message', message.idForLogging());
+      // TODO: Is `ev.confirm` a `Promise`? Original code didn’t return it:
+      return ev.confirm();
     }
 
+    const { type, id } = data.message.group ?
```

Comment:
Any chance you'd be willing to write this in the form:

```
const blah = predicate
  ? trueResult
  : falseResult;
```

I feel like that's easier to read. It's a bit of a challenge already with that destructuring... :0/

Reply:
Works for me! Added `operator-linebreak` ESLint rule for this: 7f6c856009dfce65ba42bfb301af2ed05a43ffcb